data/core/commands/core.lua

```diff
@@ -91,6 +91,8 @@ command.add(nil, {
 
   ["core:new-doc"] = function()
     core.root_view:open_doc(core.open_doc())
+    local node = core.root_view:get_active_node()
+    node:scroll_tabs_to_visible()
```

Comment:
This should probably be done in `Node:set_active_view`.
If you do that, you'll also want to do something like:
```diff
@@ -19,15 +19,15 @@ function Node:new(type)
   self.size = { x = 0, y = 0 }
   self.views = {}
   self.divider = 0.5
-  if self.type == "leaf" then
-    self:add_view(EmptyView())
-  end
   self.hovered = {x = -1, y = -1 }
   self.hovered_close = 0
   self.tab_shift = 0
   self.tab_offset = 1
   self.tab_width = style.tab_width
   self.move_towards = View.move_towards
+  if self.type == "leaf" then
+    self:add_view(EmptyView())
+  end
 end
```
As calling `Node:get_visible_tabs_number` would result in trying to do math with `self.tab_offset` that is `nil`.